great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -1156,16 +1162,23 @@ def _build_datasource_from_config(
         return datasource
 
     def _instantiate_datasource_from_config_and_update_project_config(
-        self, name: str, config: dict, initialize: bool = True
+        self,
+        name: str,
+        config: dict,
+        initialize: bool = True,
+        save_changes: bool = False,
     ) -> Optional[Datasource]:
         """ """
         datasource_config: DatasourceConfig = datasourceConfigSchema.load(
             CommentedMap(**config)
         )
 
-        self._datasource_store.set_by_name(
-            datasource_name=name, datasource_config=datasource_config
-        )
+        if save_changes:
+            self._datasource_store.set_by_name(
+                datasource_name=name, datasource_config=datasource_config
+            )
```

Comment:
Do we also need to update the in memory config here?

Reply:
You're correct! Updated accordingly 🙇🏽